pandas/core/indexes/category.py

```diff
@@ -211,29 +211,6 @@ def __new__(
 
         return cls._simple_new(data, name=name)
 
-    def _create_from_codes(self, codes, dtype=None, name=None):
```

Comment:
rather than delete this, could this be something like
```
def _create_from_codes(self, codes, name=None):
    name = name if name else self.name
    cat = self._data._from_backing_data(codes)
    return type(self)._simple_new(cat, name=name)
```

Reply:
it only saves 1 line, and id rather have fewer constructor methods

Reply:
don't we have the same code duplicated 4 times. (except that name should be lib._no_default here)

Reply:
yes.  the ongoing process is to share more of this with the other ndarray-backed-ea-indexes

Reply:
ahh. so instead of the parameter name being `codes` it would be more generic? wouldn't this give more opportunities for de-duplication and all numpy backed indexes could have a `_create_from_codes` called say ``_from_backing_data` to be consistent with the underlying array

Reply:
im hopeful but not there yet.  theres a lot of special cases in the DTI/TDI code because of `.freq`

Reply:
yeah the purpose is to avoid special casing things as much as possible; here the constructor is special cases.